File: src/_1ProgrammingBasicsWithJavaMay2023/_2ConditionalStatements/_2Exercise/_07Shopping.java
package _1ProgrammingBasicsWithJavaMay2023._2ConditionalStatements._2Exercise;

import java.util.Scanner;

public class _07Shopping {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double budget = Double.parseDouble(scanner.nextLine());
        int videoCardCount = Integer.parseInt(scanner.nextLine());
        int processorCount = Integer.parseInt(scanner.nextLine());
        int ramCount = Integer.parseInt(scanner.nextLine());
        double videoCardPrice = videoCardCount * 250;
        double processorPrice = processorCount * (videoCardPrice * 0.35);
        double ramPrice = ramCount * (videoCardPrice * 0.10);
        double totalPrice = videoCardPrice + processorPrice + ramPrice;
        if (videoCardCount > processorCount) {
            totalPrice = totalPrice * 0.85;
        }
        double diff = Math.abs(totalPrice - budget);
        if (totalPrice <= budget) {
            System.out.printf("You have %.2f leva left!", diff);
        } else {
            System.out.printf("Not enough money! You need %.2f leva more!", diff);
        }
    }
}

File: src/_1ProgrammingBasicsWithJavaScriptNovember2024/_2ConditionalStatements/_2Exercise/_07Shopping.js
function shopping(budged, numberOfVideoCards, numberOfProcessors, numberOfRamMemory) {
    const PERCENT_DISCOUNT_IF_VIDEOCARS_IS_MORE_THAN_PROCESSORS = 0.15;

    const VIDEOCARD_PRICE = 250;
    const PROCESSOR_PRICE_PERCENT = 0.35;
    const RAM_MEMORY_PRICE_PERCENT = 0.10;

    const sumForVideoCards = numberOfVideoCards * VIDEOCARD_PRICE;
    const priceForProcessor = sumForVideoCards * PROCESSOR_PRICE_PERCENT;
    const sumForProcessors = numberOfProcessors * priceForProcessor;
    const priceForRamMemory = sumForVideoCards * RAM_MEMORY_PRICE_PERCENT;
    const sumForRamMemory = numberOfRamMemory * priceForRamMemory;
    let totalSum = sumForRamMemory + sumForProcessors + sumForVideoCards;

    if (numberOfVideoCards > numberOfProcessors) {
        totalSum = totalSum - (totalSum * PERCENT_DISCOUNT_IF_VIDEOCARS_IS_MORE_THAN_PROCESSORS);
    }

    const diff = Math.abs(totalSum - budged);
    if (budged >= totalSum) {
        console.log(`You have ${diff.toFixed(2)} leva left!`);
    } else {
        console.log(`Not enough money! You need ${diff.toFixed(2)} leva more!`);
    }
}